torch/_inductor/pattern_matcher.py

```diff
@@ -320,21 +380,27 @@ class _TargetExpr(PatternExpr):
     Base class for filtering match by node.target
     """
 
-    op: Optional[str] = None
+    fns: List[FnsType]
+    fns_set: Set[FnsType]
 
-    def __init__(self, fns, users: Union[Multiple, int] = 1):
-        if not self.op:
-            raise NotImplementedError("Shouldn't directly use _BaseNodeMatch")
+    def __init__(
+        self, fns: Union[FnsType, Sequence[FnsType]], users: Union[Multiple, int] = 1
+    ) -> None:
         super().__init__()
         fns = [fns] if callable(fns) or isinstance(fns, str) else list(fns)
-        for fn in list(fns):
+        for fn in fns:
             if isinstance(fn, torch._ops.OpOverloadPacket):
-                fns.extend([getattr(fn, overload) for overload in fn.overloads()])
+                fns.extend(getattr(fn, overload) for overload in fn.overloads())
```

Comment:
Still extend. Append would add a single entry. Since extend() is happy to take an iterator all this change does it make it not create an unnecessary temporary list.